cms/djangoapps/contentstore/api/views/course_validation.py

```diff
@@ -106,28 +106,54 @@ def _dates_validation(self, course):
             has_end_date=course.end is not None,
         )
 
-    def _assignments_validation(self, course):
+    def _assignments_validation(self, course, request):
         assignments, visible_assignments = self._get_assignments(course)
-        assignments_with_dates = [a for a in visible_assignments if a.due]
+        assignments_with_dates = [
+            a for a in visible_assignments if a.due
+        ]
 
-        num_with_dates = len(assignments_with_dates)
-        num_with_dates_after_start = (
-            len([a for a in assignments_with_dates if a.due > course.start])
+        assignments_with_dates_before_start = (
+            [
+                {'id': unicode(a.location), 'display_name': a.display_name}
+                for a in assignments_with_dates 
+                if a.due < course.start
+            ]
             if self._has_start_date(course)
-            else 0
+            else []
         )
-        num_with_dates_before_end = (
-            len([a for a in assignments_with_dates if a.due < course.end])
+
+        assignments_with_dates_after_end = (
+            [
+                {'id': unicode(a.location), 'display_name': a.display_name}
+                for a in assignments_with_dates 
+                if a.due > course.end
+            ]
             if course.end
-            else 0
+            else []
         )
 
+        if get_bool_param(request, 'graded_only', False):
+            assignments_with_dates = [
+                a
+                for a in visible_assignments 
+                if a.due and a.graded
+            ]
+
+            assignments_with_dates_before_start = [
+                {'id': unicode(a.location), 'display_name': a.display_name}
+                for a in assignments_with_dates if a.due < course.start
+            ]
+
+            assignments_with_dates_after_end = [
+                {'id': unicode(a.location), 'display_name': a.display_name}
+                for a in assignments_with_dates if a.due > course.end
+            ]
+
         return dict(
             total_number=len(assignments),
             total_visible=len(visible_assignments),
-            num_with_dates=num_with_dates,
-            num_with_dates_after_start=num_with_dates_after_start,
-            num_with_dates_before_end=num_with_dates_before_end,
+            assignments_with_dates_before_start=assignments_with_dates_before_start,
```

Comment:
@fysheets I'm not sure whether we need a list of all the courses with dates or list of all the courses without dates (following the pattern of `num_with_dates`) from before. I figure we can add it in later if we need it. Any thoughts?

Reply:
If you don't think we need it currently or near term, I think it's fine to add it when it's necessary. When it becomes necessary - it seems like it would return a fairly large list, so I could see it as a field that gets added into the API but is also configurable on whether or not to include it in the response as we do with other fields.